### PR TITLE
Even lighter docker image.

### PR DESCRIPTION
**Pull request by Natim**
@leplatrem r?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+ADD CHANGELOG.rst README.rst CONTRIBUTORS.rst setup.py MANIFEST.in /code/
 ENV KINTO_INI /etc/kinto/kinto.ini
-RUN kinto --ini $KINTO_INI --backend=memory init
-CMD kinto --ini $KINTO_INI migrate && kinto --ini $KINTO_INI start
+
+# Install build dependencies, build the virtualenv and remove build
+# dependencies all at once to build a small image.
+RUN \
+    apt-get update; \
+    apt-get install -y python3 python3-pip python3-venv git build-essential make; \
```
Comment (leplatrem) [inline Dockerfile]: nit: make is already in build-essential

nit2: why git ?

File: Dockerfile [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+    apt-get install -y python3-dev libssl-dev libffi-dev libpq5 libpq-dev; \
```
Comment (leplatrem) [inline Dockerfile]: nit: libpq5 is redundant with libpq-dev

Comment (leplatrem) [inline Dockerfile]: nit2: python3 is redundant with python3-dev

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+    python3 -m venv /home/kinto; \
+    /home/kinto/bin/pip install -U pip; \
+    /home/kinto/bin/pip install cliquet[postgresql,monitoring]; \
+    /home/kinto/bin/pip install -e /code; \
+    /home/kinto/bin/kinto --ini $KINTO_INI --backend=memory init; \
+    apt-get remove -y -qq git build-essential git make python3-pip python3-venv libssl-dev libffi-dev libpq-dev python3-dev; \
```
Comment (leplatrem) [inline Dockerfile]: Add a comment like: `Uninstall from container every lib that are used for installation only`

File: Dockerfile (hunk 2 above)
Comment (Natim) [inline Dockerfile]: For both ``libpq5`` and ``python3`` it is to mark them as manually installed in order for them not to be uninstalled with removing ``python3-dev`` and ``libpq-dev``

File: Dockerfile (hunk 1 above)
Comment (Natim) [inline Dockerfile]: git is in case there are dependency link

File: Dockerfile [hunk 4]
```diff
@@ -1,13 +1,24 @@
...
+ADD CHANGELOG.rst README.rst CONTRIBUTORS.rst setup.py MANIFEST.in /code/
 ENV KINTO_INI /etc/kinto/kinto.ini
-RUN kinto --ini $KINTO_INI --backend=memory init
+
+# Install build dependencies, build the virtualenv and remove build
+# dependencies all at once to build a small image.
+RUN \
+    apt-get update; \
+    apt-get install -y python3 libpq5; \
+    apt-get install -y build-essential git python3-dev python3-setuptools libssl-dev libffi-dev libpq-dev; \
```
Comment (leplatrem) [inline Dockerfile]: micronit: you don't uninstall setuptools :)

**Comment by leplatrem** [comment]
r+!

**Comment by leplatrem** [comment]
from what I understand we can delete the kinto-docker repo after that

File: Dockerfile (hunk 4 above)
Comment (Natim) [inline Dockerfile]: And I don't uninstall pip, should I?

Comment (Natim) [inline Dockerfile]: That's about it:

```
  File "/usr/local/bin/kinto", line 5, in <module>
    from pkg_resources import load_entry_point
ImportError: No module named 'pkg_resources'
```

Comment (leplatrem) [inline Dockerfile]: ah oh ok :)

nanonit: you can put it on the same line as python3 and libpq5 then :)

File: Dockerfile [hunk 5]
```diff
@@ -1,13 +1,24 @@
 # Mozilla Kinto server
-FROM python:3.5
+FROM debian:sid
```
Comment (Natim) [inline Dockerfile]: stackbrew/debian    sid                 e560b1d2575a        8 days ago          117.2 MB
debian              sid                 e560b1d2575a        8 days ago          117.2 MB

Look at the IMAGE ID, they are the exact same images :)

**Pull request merged**